pkg/ui/handlers.go

```diff
@@ -1,23 +1,47 @@
 package ui
 
 import (
+	"fmt"
 	"io/fs"
 	"net/http"
 	"path"
+	"regexp"
+	"strings"
 
 	"github.com/canonical/identity-platform-login-ui/internal/logging"
 	"github.com/go-chi/chi/v5"
 )
 
 type API struct {
 	fileServer http.Handler
-
-	logger logging.LoggerInterface
+	fileSystem fs.FS
+	logger     logging.LoggerInterface
 }
 
 func (a *API) RegisterEndpoints(mux *chi.Mux) {
 	// TODO @shipperizer unsure if we deal with any POST/PUT/PATCH via js
-	mux.HandleFunc("/*", a.uiFiles)
+	for _, route := range a.uiRoutes() {
+		mux.HandleFunc(route, a.uiFiles)
+	}
+}
+
+func (a *API) uiRoutes() []string {
+	routes := make([]string, 0)
+	nodeList, _ := fs.ReadDir(a.fileSystem, ".")
+
+	regex_html, _ := regexp.Compile(`html$`)
+	regex_misc, _ := regexp.Compile(`(svg|txt)$`)
```

Comment:
I would put this in the constructor

at the same time u can bundle these into one, everything u put inside the round brackets can be fetched via a match result so you could use it in the if statement to decide how to strip the route path

Reply:
ok